servers/src/main/java/tachyon/master/lineage/checkpoint/CheckpointLatestScheduler.java

```diff
@@ -25,21 +32,32 @@
  * class serves as an example to implement a scheduler.
  */
 public final class CheckpointLatestScheduler implements CheckpointScheduler {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
   /**
    * CheckpointLatestScheduler does not use the lineage store view.
    *
-   * @param storeView view of a lineage store
+   * @param lineageStoreView view of a lineage store
+   * @param fileSystemMasterView a view of the file system master
    */
-  public CheckpointLatestScheduler(LineageStoreView storeView) {}
+  public CheckpointLatestScheduler(LineageStoreView lineageStoreView,
+      FileSystemMasterView fileSystemMasterView) {}
 
   @Override
-  public CheckpointPlan schedule(LineageStoreView store) {
+  public CheckpointPlan schedule(LineageStoreView store,
+      FileSystemMasterView fileSystemMasterView) {
     Lineage toCheckpoint = null;
     long latestCreated = 0;
     for (Lineage lineage : store.getAllLineagesInTopologicalOrder()) {
-      if (!lineage.isCompleted() || lineage.isPersisted() || lineage.needRecompute()
-          || lineage.isInCheckpointing()) {
+      try {
+        if (!LineageStateUtils.isCompleted(lineage, fileSystemMasterView)
+            || LineageStateUtils.isPersisted(lineage, fileSystemMasterView)
+            || LineageStateUtils.needRecompute(lineage, fileSystemMasterView)
+            || LineageStateUtils.isInCheckpointing(lineage, fileSystemMasterView)) {
+          continue;
+        }
+      } catch (FileDoesNotExistException e) {
+        LOG.error("The lineage file does not exist {}", e);
```

Comment:
This will call http://www.slf4j.org/api/org/slf4j/Logger.html#error(java.lang.String, java.lang.Throwable), not replacing the "{}". When the last argument to LOG.x is a Throwable it isn't used for the format string and is instead printed with stacktrace on the next line.